assets/js/SignGroups.tsx

```diff
@@ -1,7 +1,361 @@
+                      </div>
+                      <div className="sign_groups--select_signs-zone_col-default">
+                        {(['c', 'm'] as Zone[]).map((zone) => (
+                          <ZoneSelector
+                            key={`${station}-${zone}`}
+                            config={station}
+                            zone={zone}
+                            line={line}
+                            selectedSigns={selectedSigns}
+                            onSignChange={onSignChange}
+                            readOnly={readOnly}
+                            kind="default"
+                          />
+                        ))}
+                      </div>
+                      <div className="sign_groups--select_signs-zone_col-default">
+                        {(['e', 'n'] as Zone[]).map((zone) => (
+                          <ZoneSelector
+                            key={`${station}-${zone}`}
+                            config={station}
+                            zone={zone}
+                            line={line}
+                            selectedSigns={selectedSigns}
+                            onSignChange={onSignChange}
+                            readOnly={readOnly}
+                            kind="default"
+                          />
+                        ))}
+                      </div>
+                    </div>
+                    <div className="sign_groups--select_signs-zone_col-named">
+                      {(['n', 's', 'e', 'w', 'c', 'm'] as Zone[]).map(
+                        (zone) => (
+                          <ZoneSelector
+                            key={`${station}-${zone}`}
+                            config={station}
+                            zone={zone}
+                            line={line}
+                            selectedSigns={selectedSigns}
+                            onSignChange={onSignChange}
+                            readOnly={readOnly}
+                            kind="named"
+                          />
+                        ),
+                      )}
+                    </div>
+                  </div>
+                </div>
+              </div>
+            ))}
+          </div>
+        </div>
+        <div className="sign_groups--container-right">
+          <div>
+            Set Custom Message
+            <div className="sign_groups--custom_message-container">
+              <div>
+                <CustomTextInput
+                  signID="sign_group"
+                  line1={line1}
+                  line2={line2}
+                  onValidLine1Change={setLine1}
+                  onValidLine2Change={setLine2}
+                />
+              </div>
+              <div className="sign_groups--virtual_sign">
+                <VirtualSign line1={line1} line2={line2} time={currentTime} />
+              </div>
+            </div>
+          </div>
+          <div>
+            Schedule return to &quot;Auto&quot;
```

Comment:
Is the `&quot;` actually needed in a text node? I don't think it would otherwise be a special character here.

Reply:
Yeah, otherwise runs afoul of `react/no-unescaped-entities`.